pkg/expression/builtin_string.go

```diff
@@ -3377,11 +3380,17 @@ func (b *builtinFormatWithLocaleSig) evalString(ctx EvalContext, row chunk.Row)
 	tc := typeCtx(ctx)
 	if isNull {
 		tc.AppendWarning(errUnknownLocale.FastGenByArgs("NULL"))
-	} else if !strings.EqualFold(locale, "en_US") { // TODO: support other locales.
-		tc.AppendWarning(errUnknownLocale.FastGenByArgs(locale))
 	}
-	locale = "en_US"
-	formatString, err := mysql.GetLocaleFormatFunction(locale)(x, d)
+
+	lang, err := language.Parse(locale)
+	if err != nil {
+		return "", false, err
+	}
+	p := message.NewPrinter(lang)
+	xint, _ := strconv.ParseFloat(x, 64)
```

Comment:
Why ignore error here? I think it's better to handle error even if we suppose it should not happen